### DISCUSSION TITLE
主备模式删除topic，从节点物理consumequeue不删除

### DISCUSSION THREAD

**Discussion by xinjianxun** [apache/rocketmq#7058] [Q&A] [answered]
主备模式或者主备自动切换模式下，删除topic，主节点consumequeuq下的物理文件删除，从节点却保留。这样从节点切换回主节点后再次创建这个topic后会发现消息依然存在，请问这么设计的意图或者是个bug？

**Comment by RongtongJin** [comment, marked as answer]
我认为这是不合理的。当前删除topic操作只在主上面，然后通过主备同步元数据，将备中的元数据删除。但consumequeue的文件不会被删除（需要等到过期清理流程进行删除），这造成上述问题。这个需要进行优化。

**Comment by frinda** [comment]
https://github.com/apache/rocketmq/pull/7273

**Reply by frinda** [reply]
这个MR解决此问题，但是没人给合入
